test/e2e/healthchek.go

```diff
@@ -23,94 +25,120 @@ const (
-	controlplaneIP, err := MachineToIPAddress(ctx, cli, &controlplaneMachine, baremetalv4Pool[0])
+	controlplaneMachine := GetMachine(ctx, bootstrapClusterClient, client.ObjectKey{Name: controlplaneMachineName, Namespace: namespace})
+	controlplaneIP, err := MachineToIPAddress(ctx, bootstrapClusterClient, &controlplaneMachine, baremetalv4Pool[0])
 	Expect(err).ToNot(HaveOccurred())
-	Expect(runCommand("", "", controlplaneIP, "metal3", "systemctl stop kubelet")).To(Succeed())
+
+	Logf("Stopping kubelet on controlplane machine")
+	Eventually(func(g Gomega) {
+		g.Expect(runCommand("", "", controlplaneIP, "metal3", "systemctl stop kubelet")).To(Succeed())
+	}, 2*time.Minute, 10*time.Second).Should(Succeed())
+
 	// Wait until node is marked unhealthy and then check that it becomes healthy again
 	Logf("Waiting for unhealthy controlplane ...")
-	WaitForHealthCheckCurrentHealthyToMatch(ctx, cli, 2, controlplaneHealthcheck, timeout, freq)
+	WaitForHealthCheckCurrentHealthyToMatch(ctx, bootstrapClusterClient, 2, controlplaneHealthcheck, timeout, freq)
 	Logf("Waiting for remediationrequest to exist ...")
-	WaitForRemediationRequest(ctx, cli, client.ObjectKeyFromObject(&controlplaneMachine), true, timeout, freq)
+	WaitForRemediationRequest(ctx, bootstrapClusterClient, client.ObjectKeyFromObject(&controlplaneMachine), true, timeout, freq)
 	Logf("Waiting for controlplane to be healthy again...")
-	WaitForHealthCheckCurrentHealthyToMatch(ctx, cli, 3, controlplaneHealthcheck, timeout, freq)
+	WaitForHealthCheckCurrentHealthyToMatch(ctx, bootstrapClusterClient, 3, controlplaneHealthcheck, timeout, freq)
 	Logf("Waiting for remediationrequest to not exist ...")
-	WaitForRemediationRequest(ctx, cli, client.ObjectKeyFromObject(&controlplaneMachine), false, timeout, freq)
+	WaitForRemediationRequest(ctx, bootstrapClusterClient, client.ObjectKeyFromObject(&controlplaneMachine), false, timeout, freq)
+
+	By("Deleting worker and controlplane Metal3RemediationTemplate CRs")
+	Expect(bootstrapClusterClient.Delete(ctx, workerRemediationTemplate)).To(Succeed(), "should delete worker Metal3RemediationTemplate CR")
+	Expect(bootstrapClusterClient.Delete(ctx, controlplaneRemediationTemplate)).To(Succeed(), "should delete controlplane Metal3RemediationTemplate CR")
+
+	By("Make sure Metal3RemediationTemplate CRs are actually deleted")
+	Eventually(func() bool {
+		cpM3MremediationTemplate := &infrav1.Metal3RemediationTemplate{}
+		workerM3MremediationTemplate := &infrav1.Metal3RemediationTemplate{}
+		cpErr := bootstrapClusterClient.Get(ctx, client.ObjectKeyFromObject(controlplaneRemediationTemplate), cpM3MremediationTemplate)
+		workerErr := bootstrapClusterClient.Get(ctx, client.ObjectKeyFromObject(workerRemediationTemplate), workerM3MremediationTemplate)
+		return apierrors.IsNotFound(cpErr) && apierrors.IsNotFound(workerErr)
+	}, 2*time.Minute, 10*time.Second).Should(BeTrue(), "Metal3RemediationTemplate should have been deleted")
```

Comment:
Should these intervals be added to the e2e_conf.yaml instead?

Reply:
Could you reference them here, instead of hardcoding?

Reply:
sorry it was missed, now fixed